### PR TITLE
Add Board class and specs for it

### PR DESCRIPTION
**Pull request by ilasorsa**

### CODE REVIEW

**Comment by matugm** [comment]
`expect(board.is_empty?).to be true` -> `expect(board).to be_empty`

**Comment by ilasorsa** [comment]
You're right. I fixed them :)

File: lib/board.rb [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+    def insert_into_column(column, color)
+      column -= 1
+      grid[first_free_row_for column][column] = color
+    end
+
+    private
+
+    def first_free_row_for(column)
+      ROWS.times { |row| return row if grid[row][column].nil? }
```
Comment (ashleymichal) [inline lib/board.rb]: We discussed it during Ruby Monday, and we believe it would be easier to re-work the `grid` data structure to list columns of rows rather than rows of columns so that this line could instead become

``` ruby
column.index(nil)
```

which would find the first empty slot in the column.

File: spec/game_spec.rb [hunk 2]
```diff
@@ -1,17 +1,19 @@
...
-describe Game do
-  let(:game) { Game.new }
+module RubyConnect
+  describe Game do
+    let(:game) { subject }
 
-  it 'has an empty board' do
-    expect(game.board).to have_attributes(size: 7)
-  end
+    it 'has an empty board' do
```
Comment (matugm) [inline spec/game_spec.rb]: We are also testing for empty board on the board specs, should be delete one of the test?

File: spec/board_spec.rb [hunk 3]
```diff
@@ -0,0 +1,40 @@
...
+  describe Board do
+    let(:board) { subject }
+
+    context 'when a move is still not done' do
+      it 'is empty' do
+        expect(board).to be_empty
+      end
+    end
+
+    context 'when a move is done' do
```
Comment (matugm) [inline spec/board_spec.rb]: What does it means for a move to 'be done'? I think we could rename this to 'after a player selects a column'.

**Comment by ilasorsa** [comment]
I made changes discussed in the last hangout. I'm sorry for the English, I'm not a native language speaker: feel free to change every text in the specs. Same for the code, sure :)

File: lib/board.rb [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+      grid[column][row] = color
+    end
+
+    private
+
+    def first_slot(column)
+      grid[column].index(nil) || raise(ArgumentError, "Column full")
+    end
+
+    def elements
```
Comment (ashleymichal) [inline lib/board.rb]: 'elements' is a rather cryptic name for this. Perhaps ''slots' to match the 'first_slot' method? elements makes me think Array. I also agree with @matugm's comment in the Slack channel that 'first_open_slot' was more descriptive than 'first_slot'
